src/options.js

```diff
@@ -68,8 +68,17 @@ function maybeTypedArrowify(vector, type) {
   return vector == null
     ? vector
     : (type === undefined || type === Array) && isArrowDateType(vector.type)
-    ? coerceDates(vector.toArray())
-    : maybeTypedArrayify(vector.toArray(), type);
+    ? coerceDates(vectorToArray(vector))
+    : maybeTypedArrayify(vectorToArray(vector), type);
+}
+
+function vectorToArray(vector) {
+  const values = vector.toArray();
+  if (values.nullCount) {
+    const NULL = vector.type?.typeId === 5 /* Utf8 */ ? null : undefined;
+    for (let i = 0; i < values.length; ++i) if (!values.isValid(i)) values[i] = NULL;
```

Comment:
We can’t use assignment (mutation) here because this might not be a copy. According to [the docs](https://arrow.apache.org/docs/js/classes/Arrow_dom.Vector.html#toArray):

> If this Vector contains a single Data chunk and the Vector's type is a primitive numeric type corresponding to one of the JavaScript TypedArrays, this method returns a zero-copy slice of the underlying TypedArray values. If there's more than one chunk, the resulting TypedArray will be a copy of the data from each chunk's underlying TypedArray values.